keys/keys.go

```diff
@@ -369,3 +371,251 @@ func Range(ba roachpb.BatchRequest) roachpb.RSpan {
+		{name: "Transaction", suffix: localTransactionSuffix, atEnd: false},
+	}
+
+	for _, s := range suffixDict {
+		if s.atEnd {
+			if bytes.HasSuffix(key, s.suffix) {
+				key = key[:len(key)-len(s.suffix)]
+				fmt.Fprintf(&buf, "/%s%s", s.name, decodeKeyPrint(key))
+				return buf.String()
+			}
+		} else {
+			begin := bytes.Index(key, s.suffix)
+			if begin > 0 {
+				addrKey := key[:begin]
+				id := key[(begin + len(s.suffix)):]
+				fmt.Fprintf(&buf, "/%s/addrKey:%s/id:%s", s.name, decodeKeyPrint(addrKey), id.String())
+				return buf.String()
+			}
+		}
+	}
+	fmt.Fprintf(&buf, "%s", decodeKeyPrint(key))
+	return buf.String()
+}
+
+func hexPrint(key roachpb.Key) string {
+	return fmt.Sprintf("/%s", key.String())
+}
+
+// TableDataPrefix
+func decodeKeyPrint(key roachpb.Key) string {
+	var buf bytes.Buffer
+	for k := 0; len(key) > 0; k++ {
+		var d interface{}
+		var err error
+		switch encoding.PeekType(key) {
+		case encoding.Null:
+			key, _ = encoding.DecodeIfNull(key)
+			d = parser.DNull
+		case encoding.NotNull:
+			key, _ = encoding.DecodeIfNotNull(key)
+			d = "#"
+		case encoding.Int:
+			var i int64
+			key, i, err = encoding.DecodeVarint(key)
+			d = parser.DInt(i)
+		case encoding.Float:
+			var f float64
+			key, f, err = encoding.DecodeFloat(key, nil)
+			d = parser.DFloat(f)
+		case encoding.Bytes:
+			var s string
+			key, s, err = encoding.DecodeString(key, nil)
+			d = parser.DString(s)
+		case encoding.Time:
+			var t time.Time
+			key, t, err = encoding.DecodeTime(key)
+			d = parser.DTimestamp{Time: t}
+		default:
+			// This shouldn't ever happen, but if it does let the loop exit.
+			d = key.String()
+			key = nil
+		}
+
+		if err != nil {
+			fmt.Fprintf(&buf, "/<%v>", err)
+			continue
+		}
+		fmt.Fprintf(&buf, "/%s", d)
+	}
+	return buf.String()
+}
+
+// PrettyPrint print the key with more human readable, which organized as follow
+func PrettyPrint(key roachpb.Key) string {
+	keyDict := []struct {
+		name    string
+		start   roachpb.Key
+		end     roachpb.Key
+		entries []dictEntry
+	}{
+		{name: "Local", start: localPrefix, end: LocalMax, entries: []dictEntry{
+			{name: "Store", prefix: roachpb.Key(localStorePrefix), ppFunc: localStoreKeyPrint},
+			{name: "RangeID", prefix: roachpb.Key(LocalRangeIDPrefix), ppFunc: localRangeIDKeyPrint},
+			{name: "Range", prefix: LocalRangePrefix, ppFunc: localRangeKeyPrint},
+		}},
+		{name: "System", start: SystemPrefix, end: SystemMax, entries: []dictEntry{
+			{name: "Meta2", prefix: Meta2Prefix, ppFunc: hexPrint},
+			{name: "Meta1", prefix: Meta1Prefix, ppFunc: hexPrint},
+			//{name: "Meta", prefix: MetaPrefix},
+			{name: "StatusStore", prefix: StatusStorePrefix, ppFunc: decodeKeyPrint},
+			{name: "StatusNode", prefix: StatusNodePrefix, ppFunc: decodeKeyPrint},
+			//{name: "Status", prefix: StatusPrefix},
+		}},
+		{name: "User", start: TableDataPrefix, end: nil, entries: []dictEntry{
+			{name: "TableData", prefix: TableDataPrefix, ppFunc: decodeKeyPrint},
```

Comment:
All user data is table data and that is unlikely to change any time soon. Can we shorten from `/User/TableData` to `/Table`?